### DISCUSSION TITLE
Schedule stop pair

### DISCUSSION THREAD

**Discussion by adrianleh** [transitland/transitland#396] [Q&A] [answered]
Hi,

I'm trying to use an older project built on transitland (https://github.com/transitland/transitland-processing-animation). It uses the v1 `schedule_stop_pairs` endpoint, which seems to be returning empty arrays only. Has this endpoint been deprecated?

e.g. `http://transit.land/api/v1/schedule_stop_pairs?operator_onestop_id=o-9q9-bart&date=2023-04-17&per_page=10000&sort_min_id=0&apikey=(REDACTED)`

**Comment by drewda** [comment, marked as answer]
Yes, the v1 Datastore API has been deprecated. See https://www.transit.land/documentation/#comparing-the-apis

For the time being, a number of the v1 API endpoints do continue to work, as they are served behind the scenes with a temporary fix that powers them using v2 data. However, the schedule stop pairs endpoint is not one that can be served by v2 data. The team has occasionally maintained the schedule stop pair endpoint, but is not able to making any fixes at this point.

The v2 REST API has much of the same functionality, just in a slightly different format, available through its stop departures endpoint: https://www.transit.land/documentation/rest-api/departures This endpoint is fully supported currently.

**Reply by adrianleh** [reply]
Hi! Thanks for getting back to me and pointing me to the new endpoints! As a follow-up question: Are you planning on updating your visualization project I linked above to use the new API?

**Reply by drewda** [reply]
The core Transitland team has no plans to work on the [transitland-processing-animation](https://github.com/transitland/transitland-processing-animation) project. But we're glad to take questions on this discussion forum from anyone who wants to work on it, similar to other Transitland API consumers.
